sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogUtils.scala

```diff
@@ -280,7 +281,15 @@ object CatalogUtils {
    * @return the URI of the path
    */
   def stringToURI(str: String): URI = {
```

Comment:
This caused a regression in our environment, because it's a low-level API that is used in many places. The problematic one is in `HiveClientImpl` where we get the schema/table location string from HMS. We shouldn't perform the empty string check if that malformed location string is already in HMS. Spark should still be able to get it and display it in places like `DESCRIBE` command.

I think it's better to perform validations explicitly, instead of in a API like `stringToURI` which should simply perform the conversion.

Shall we revert this change and revisit it?  @yaooqinn

Reply:
It looks ok to me to 'display' the invalid path in either errors or command outputs. The error at least allows users to fix malformed location infomation

Reply:
Error message is not a "display". Assume there is a SQL UI product relying on DESCRIBE commands to get metadata, this change will break the product.

Reply:
They can do their error handling by themselves. We have a lot of cases to fail such a case，for example， desc or table with invalid hive serde settings

Reply:
Let's be more serious about breaking changes. Yes, we can ask the downstream applications to work around breaking changes, but this has cost and we should do it with caution.

I don't think a refactor/code cleanup is justified for a breaking change (and there is no legacy config to restore). I suggest we revert it and use a more explicit way to perform the empty location check for places that need this check.

cc @dongjoon-hyun @gatorsmile